### PR TITLE
Fix audit check for synapse-rpc and chrono crate

### LINKED ISSUE #271: Fix Synapse -> rpc cargo audit issues which are blocking Merge
```
Calling cargo-audit (JSON output)
  /home/runner/.cargo/bin/cargo audit --json
  {"database":{"advisory-count":393,"last-commit":"9839c6ee0fdc486eea186f36ee8efd1f32d368f2","last-updated":"2022-01-24T14:39:43Z"},"lockfile":{"dependency-count":443},"settings":{"target_arch":null,"target_os":null,"severity":null,"ignore":["RUSTSEC-2020-0159"],"informational_warnings":["unmaintained"],"package_scope":null},"vulnerabilities":{"found":true,"count":1,"list":[{"advisory":{"id":"RUSTSEC-2022-0006","package":"thread_local","title":"Data race in `Iter` and `IterMut`","description":"In the affected version of this crate, `{Iter, IterMut}::next` used a weaker memory ordering when loading values than what was required, exposing a potential data race\nwhen iterating over a `ThreadLocal`'s values.\n\nCrates using `Iter::next`, or `IterMut::next` are affected by this issue.","date":"2022-01-23","aliases":[],"related":[],"collection":"crates","categories":["memory-corruption"],"keywords":[],"cvss":null,"informational":null,"url":"https://github.com/Amanieu/thread_local-rs/issues/33","references":[],"withdrawn":null},"versions":{"patched":[">=1.1.4"],"unaffected":[]},"affected":null,"package":{"name":"thread_local","version":"0.3.6","source":"registry+https://github.com/rust-lang/crates.io-index","checksum":"c6b53e329000edc2b34dbe8545fd20e55a333362d0a321909685a19bd28c3f1b","dependencies":[{"name":"lazy_static","version":"1.4.0","source":"registry+https://github.com/rust-lang/crates.io-index"}],"replace":null}}]},"warnings":{}}
Warning: 1 vulnerabilities found!
```

### PR DESCRIPTION
**Pull request by tiainen**
Closes #271 by:

- replacing the [chrono crate](https://crates.io/crates/chrono) with the [time crate](https://crates.io/crates/time)
- use a forked synapse-rpc with fixed audit checks: https://github.com/tiainen/synapse/tree/pyrsia-synapse

### CODE REVIEW

**Review by efrisch — DISMISSED** [review]
👍

**Review by mgrand — APPROVED** [review]

**Review by fishseabowl — DISMISSED** [review]
it looks fine to me. thanks

File: Cargo.toml [hunk 1]
```diff
@@ -33,13 +33,13 @@ pretty_env_logger = "0.4.0"
...
 serde = { version = "1.0", features = ["derive"] }
-serde_json = "1.0.73"
+serde_json = "1.0.78"
 serial_test = "0.5.1"
 sha2 = { version = "0.9.8" }
 stringreader = "0.1.1"
 strum = "0.22.0"
 strum_macros = "0.22.0"
-synapse-rpc = "0.1.20"
+synapse-rpc = { git = "https://github.com/tiainen/synapse", branch = "pyrsia-synapse" }
```
Comment (erwin1) [inline Cargo.toml:42]: Once https://github.com/pyrsia/synapse/pull/1 is merged, this should be changed:

```suggestion
synapse-rpc = { git = "https://github.com/pyrsia/synapse", branch = "master" }
```

**Review by mgrand — APPROVED** [review]

File: Cargo.lock [hunk 2]
```diff
@@ -3821,17 +3775,15 @@ dependencies = [
 
 [[package]]
 name = "synapse-rpc"
-version = "0.1.20"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "766063cc2785f4311ecd10c5098f7546f26a26039ebeb3fcaea9fd067afbd95b"
+version = "0.1.21"
+source = "git+https://github.com/tiainen/synapse?branch=pyrsia-synapse#b1420d11aa9a3eae8df5f71f8324a743c6da150e"
```
Comment (erwin1) [inline Cargo.lock:3779]:
```suggestion
source = "git+https://github.com/pyrsia/synapse?branch=master#ae7429a1707247bbe7cf9d0b12950d4bb7e112ba"
```

**Review by erwin1 — APPROVED** [review]

**Review by mgrand — APPROVED** [review]

**Pull request merged**